### ISSUE TITLE
Bert-Tiny model demo results don't match the dataset reference despite good PCC

### ISSUE THREAD

**Issue by vigneshkeerthivasanx** [tenstorrent/tt-metal#14146]
Bert-Tiny model is completed and the PCC for all sub-modules > 0.99. But the PCC of the entire model has dropped after the final linear layer to ~0.96 and ~0.97 after splitting start_logits and end_logits respectively.
The models accuracy for 100 samples are as follows:

Accuracy between TTNN model and SQUADv2 reference:
```
Eval: exact: 14.423076923076923 
F1: 22.086671341479036

```
Accuracy between PyTorch model and SQUADv2 reference:
```
exact: 19.23076923076923 
F1: 24.28325947556717
```

Accuracy between TTNN and PyTorch model:
```
exact: 58.65384615384615
F1: 0.5875457875457875
```

Pytorch results does not align with the dataset for most of the cases.
However, Pytorch and TTNN results match in most cases, and the results can be found here: [Link](https://docs.google.com/spreadsheets/d/1dBKFjR5tvsdn3oCvyMyOX1D_z_LwMWEI-q7MSQjvo_o/edit?gid=1101027898#gid=1101027898)
Corresponding PR: #13471

**Comment by saichandax** [comment]
Closing the issue for now. Will re-open as per priority.
